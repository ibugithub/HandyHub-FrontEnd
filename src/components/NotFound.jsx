import React from 'react';
import { Link } from 'react-router-dom';

const NotFound = () => (
  <div className="flex flex-col justify-center items-center mx-auto">
    <p className="text-2xl">Redirecting.....</p>
    <Link to="/">Go back to home</Link>
  </div>
);

export default NotFound;
