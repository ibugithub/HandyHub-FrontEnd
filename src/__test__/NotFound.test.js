import React from 'react';
import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import NotFound from '../components/NotFound';

describe('NotFound Component', () => {
  it('renders the "Redirecting....." text', () => {
    render(
      <MemoryRouter>
        <NotFound />
      </MemoryRouter>,
    );

    const redirectingText = screen.getByText('Redirecting.....');
    expect(redirectingText).toBeInTheDocument();
  });

  it('renders a link to go back to home', () => {
    render(
      <MemoryRouter>
        <NotFound />
      </MemoryRouter>,
    );

    const linkToHome = screen.getByText('Go back to home');
    expect(linkToHome).toBeInTheDocument();
  });

  it('matches snapshot', () => {
    const { container } = render(
      <MemoryRouter>
        <NotFound />
      </MemoryRouter>,
    );

    expect(container).toMatchSnapshot();
  });
});
