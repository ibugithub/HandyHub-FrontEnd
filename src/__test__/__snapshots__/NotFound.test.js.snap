// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound Component matches snapshot 1`] = `
<div>
  <div
    class="flex flex-col justify-center items-center mx-auto"
  >
    <p
      class="text-2xl"
    >
      Redirecting.....
    </p>
    <a
      href="/"
    >
      Go back to home
    </a>
  </div>
</div>
`;
